workflow_scripts/check_model.py

```diff
@@ -0,0 +1,24 @@
+import ort_test_dir_utils
+import onnxruntime
+import onnx
+import test_utils
+
+
+def by_onnx(model_path, model_name):
+    model = onnx.load(model_path)
+    onnx.checker.check_model(model)
+    print('[PASS] {} is checked by onnx. '.format(model_name))
+
+def by_onnxruntime(model_path, model_name, test_data_set):
+    # if 'test_data_set_N' doesn't exist, create test_dir
+    if not test_data_set:
+        onnxruntime.InferenceSession(model_path)
+        ort_test_dir_utils.create_test_dir(model_path, './', test_utils.TEST_ORT_DIR)
+        ort_test_dir_utils.run_test_dir(test_utils.TEST_ORT_DIR)
+    # otherwise use the existing 'test_data_set_N' as test data
+    else:
+        test_dir_from_tar = test_utils.get_model_directory(model_path)
+        ort_test_dir_utils.run_test_dir(test_dir_from_tar)
```

Comment:
run_test_dir does not return an error in the case when expected outputs dont match actual output from the ort run. It simply prints a statement to stdout. This script won't catch this failure. You may have to update the run_test_dir script in ort to return a failure.